### PR TITLE
Updated naming and data product versioning conventions

### PR DESCRIPTION
**Pull request by bourque**
# Change Summary

## Overview

This PR updates the documents for filenaming and data product versioning conventions based on the conversations during the SDWG meeting on 03/07/2024.

## Updated Files

- `docs/source/development-guide/style-guide/naming-conventions.rst`
   - Updated filenaming conventions
- `docs/source/development-guide/style-guide/versioning.rst`
   - Update data product versioning conventions

### CODE REVIEW

File: docs/source/development-guide/style-guide/naming-conventions.rst [hunk 1]
```diff
@@ -31,46 +30,49 @@ The general filename convention is as follows::
...
+  descriptors for each instrument:
 
   * CoDICE: ``hskp``, ``eng``, ``lo-priority``, ``lo-sw-angular``, ``lo-nsw-angular``, ``lo-pha``, ``hi-pha``,
     ``lo-sw-species``, ``lo-nsw-species``, ``hi-omni``, ``hi-sectored``
-  * GLOWS: ``histogram``, ``de``, ``<pointing number>``
+  * GLOWS: ``histogram``, ``de``
   * HIT: TBD
   * IDEX: TBD
-  * IMAP-Hi: TBD
+  * IMAP-Hi: ``45sensor-counts``, ``90sensor-counts``, ``45sensor-flux-3month-map`` (more TBD)
```
Comment (greglucas) [inline docs/source/development-guide/style-guide/naming-conventions.rst]: I forgot to bring this up on the call today, but for consistency should we put `3month` at the end?
`flux-map-3month` similar to how SWAPI  has `sci-1min` with the frequency last.
cc @tech3371

File: docs/source/development-guide/style-guide/naming-conventions.rst [hunk 2]
```diff
@@ -31,46 +30,49 @@ The general filename convention is as follows::
...
   * SWAPI: TBD
   * SWE: TBD
 
 * ``<startdate>`` is the date of the 'earliest' data within the data product, in the format of ``YYYYMMDD``.
-* ``<enddate>`` is the date of the 'latest' data within the data product if the data product spans more than one day.
-  If the data does not span multiple days, then the ``<enddate>`` will be the same as the ``<startdate>``. The format
-  is also ``YYYYMMDD``.
-* ``<version>`` stores the version of the data product in the format is ``vXX-YY``.  See the versioning conventions
-  described in the :ref:`data product versioning <Data Product Versioning>` documentation for further details.
+* ``<repointing>`` is an optional component that describes the repointing number, in the format of ``repoint-<num>``,
```
Comment (greglucas) [inline docs/source/development-guide/style-guide/naming-conventions.rst]: I think we probably want to remove the "-" between repoint and number so it is a single repoint field and the hyphen separates the date and repointing fields.
```suggestion
* ``<repointing>`` is an optional component that describes the repointing number, in the format of ``repoint<num>``,
```

**Review by greglucas — APPROVED** [review]
minor nits, otherwise I think it agrees with what we discussed. Thanks for updating this!

File: docs/source/development-guide/style-guide/naming-conventions.rst (hunk 2 above)
Comment (bourque) [reply docs/source/development-guide/style-guide/naming-conventions.rst]: Ah yes, good catch.

File: docs/source/development-guide/style-guide/naming-conventions.rst (hunk 1 above)
Comment (bourque) [reply docs/source/development-guide/style-guide/naming-conventions.rst]: That is probably a good idea!

**Pull request merged**